riscv/src/compiler.rs

```diff
@@ -200,44 +200,50 @@ pub fn compile(
             }
         });
 
-    let submachine_init = call_every_submachine(coprocessors);
+    let submachine_init = call_every_submachine(coprocessors)
+        .into_iter()
+        .map(|l| (2, l));
     let bootloader_and_shutdown_routine_lines = if with_bootloader {
-        let bootloader_and_shutdown_routine = bootloader_and_shutdown_routine(&submachine_init);
-        log::debug!("Adding Bootloader:\n{}", bootloader_and_shutdown_routine);
+        let bootloader_and_shutdown_routine = bootloader_and_shutdown_routine(submachine_init);
+        log::debug!("Adding Bootloader:\n{:?}", bootloader_and_shutdown_routine);
         bootloader_and_shutdown_routine
-            .split('\n')
-            .map(|l| l.to_string())
-            .collect::<Vec<_>>()
     } else {
-        submachine_init
+        submachine_init.collect()
     };
 
-    let mut program: Vec<String> = file_ids
+    let mut program: Vec<(u8, String)> = file_ids
         .into_iter()
-        .map(|(id, dir, file)| format!(".debug file {id} {} {};", quote(&dir), quote(&file)))
+        .map(|(id, dir, file)| {
+            (
+                2,
+                format!(".debug file {id} {} {};", quote(&dir), quote(&file)),
+            )
+        })
         .chain(bootloader_and_shutdown_routine_lines)
         .collect();
     if !data_code.is_empty() {
-        program.push("x1 <== jump(__data_init);".to_string());
+        program.push((2, "x1 <== jump(__data_init);".to_string()));
     }
     program.extend([
-        format!("// Set stack pointer\nx2 <=X= {stack_start};"),
-        "x1 <== jump(__runtime_start);".to_string(),
-        "return;".to_string(), // This is not "riscv ret", but "return from powdr asm function".
+        (2, "// Set stack pointer".to_string()),
+        (2, format!("x2 <=X= {stack_start};")),
+        (2, "x1 <== jump(__runtime_start);".to_string()),
+        (2, "return;".to_string()), // This is not "riscv ret", but "return from powdr asm function".
```

Comment:
I'm not sure all this weirdness (2, ...) everywhere is worth it

Reply:
I thought about it halfway through. Anyway, this 2 is the indentation level.

Reply:
I got that, but I don't think it's a good solution.